### ISSUE TITLE
Question about fluid_styled_content requirement

### ISSUE THREAD

**Issue by schloram** [Gernott/mask#586]
Hi,
currently I'm working on a headless TYPO3 website with `EXT:headless` installed. `EXT:fluid_syteld_content` is therefore unnecessary so I don't need to install it. But I also need `EXT:mask`. The idea is then to override the mask TypoScript as needed by `EXT:headless` to output the fields as json.

Unfortunately `mask` requires `fluid_styled_content` which means that it gets installed even though it is not needed in this case.

Couldn't the requirement of `fluid_styled_content` just be dropped? I don't really see the benefit of requiring `fluid_styled_content` because `mask` isn't even using the Layouts/Partials or the TypoScript of `fluid_styled_content`. 

AFAIK requiring `frontend` would be sufficient. Or am I missing something crucial?

regards,
R

**Comment by nhovratov** [comment]
`fluid_styled_content` is actually not a hard requirement, but it provides a TypoScript snippet for `lib.parseFunc`, which is necessary for proper richtext link parsing and such. So it is required for convenience reasons. Dropping it again would be breaking and requires a new major Mask release.

**Comment by nitori** [comment]
Hm seems like headless ships it's own parseFunc.

**Comment by nhovratov** [comment]
Yep, looks like 80% is the same as in Fluid Styled Content. The question is really whether everyone should ship his own ParseFunc. Mask also had an old copy of it until it faced some bugs, because it was outdated. This was the reason FSC was added as requirement, so I don't have to keep track of every change in this file.

**Comment by nitori** [comment]
The only thing I could think of is to not make fsc a _hard_ requirement. As different extensions can ship their own parseFunc, the only requirement should be "an extension with a lib.parseFunc definition", like fsc or headless. The bootstrap_package extension even conflicts with fsc (but it's hard to say if it even makes sense to use that one together with mask).

There is of course the "risk" that some user might forget to activate fsc?

**Comment by nhovratov** [comment]
Yes, most users "just install" Mask and expect it to work without any further steps besides maybe including the static template. If we drop the requirement, I have to document it somewhere. But most users don't read the documentation. I already can see issues like: "Links don't work in RTE with Mask elements" or similar. In all honesty, I rather have "experienced" users complain about the "unnecessary" installed fluid_styled_content extension, than unexperienced users struggling to understand why links don't work. I'm also an advocate for minimal setups and dropping dead code, but the main target audience for Mask are not hardcode TYPO3 developers who could also do everything by hand.
